### PR TITLE
Feature/deps ranges

### PR DESCRIPTION
**Pull request by memsharded**
~~Not to merge.~~

~~Working on a PoC for supporting version ranges and similar expressions.~~

Initial proposal finished, could be even merged. Please review!

### CODE REVIEW

File: conans/model/requires.py [hunk 1]
```diff
@@ -18,10 +18,19 @@ def __init__(self, conan_reference, private=False, override=False, dev=False):
                     needed for building or testing, but not affects the package hash at all
         """
         self.conan_reference = conan_reference
+        self.range_reference = conan_reference
         self.private = private
         self.override = override
         self.dev = dev
 
+    def version_range(self):
```
Comment (lasote) [inline conans/model/requires.py]: Better @property?

File: conans/model/requires.py [hunk 2]
```diff
@@ -18,10 +18,19 @@ def __init__(self, conan_reference, private=False, override=False, dev=False):
...
+        version = self.range_reference.version
+        if version.startswith("[") and version.endswith("]"):
+            return version[1:-1]
+
+    def is_resolved(self):
```
Comment (lasote) [inline conans/model/requires.py]: @property

File: conans/model/requires.py (hunk 1 above)
Comment (lasote) [inline conans/model/requires.py]: Docstrings are always welcome: # From "[>0.1,<0.3]" to ">0.1,<0.3"

Comment (memsharded) [inline conans/model/requires.py]: done, both as properties with docstrings

**Pull request merged**